string state_name
string info
